src/screens/AddCrypto/index.tsx

```diff
@@ -1,10 +1,71 @@
-import {Text, TouchableOpacity} from 'react-native';
-import React from 'react';
-
-const AddCrypto = () => (
-  <TouchableOpacity>
-    <Text> Add a Cryptocurrency </Text>
-  </TouchableOpacity>
-);
+import {Modal, TouchableOpacity, Alert} from 'react-native';
+import {useState} from 'react';
+import {CryptoInput, AddText, BackText, ModalText, Touchable} from './style';
+import {useDispatch, useSelector} from 'react-redux';
+import {AppDispatch, RootState} from '../../store';
+import {useNavigation} from '@react-navigation/native';
+import {getCrypto} from '../../store/actions';
+import theme from '../../theme';
+
+const AddCrypto = () => {
+  const [text, onChangeText] = useState('');
+  const [borderColor, setBorderColor] = useState(theme.colors.borderGray);
+  const {navigate} = useNavigation();
+
+  const cryptos = useSelector(
+    (state: RootState) => state.selectedCrypto.cryptos,
+  );
+  const dispatch = useDispatch<AppDispatch>();
+
+  const findDuplicate = () => {
+    const res = cryptos.find(e => {
+      return (
+        e.name.toLowerCase() === text.toLowerCase() ||
+        e.symbol.toLowerCase() === text.toLowerCase()
+      );
+    });
+    return res;
+  };
+
+  const handleCrypto = () => {
+    try {
+      if (!text) {
+        Alert.alert('Warning', 'Crypto is required');
+      } else if (findDuplicate()) {
+        Alert.alert('Error', 'Crypto already displayed');
+      } else {
+        dispatch(getCrypto(text));
+        onChangeText('');
+        navigate('CryptoList');
+      }
+    } catch (err) {
+      return err;
+    }
+  };
+  return (
+    <Modal animationType="fade">
+      <TouchableOpacity onPress={() => navigate('CryptoList' as never)}>
```

Comment:
as never? 
This is not a correct type, check the type configuration of your navigation